andela/travel_tool_front
# #161755766 Fix dashboard redirect issue

#### What does this PR do?
Prevents the dashboard page from redirecting to request page when reloaded

#### Description of Task to be completed?
- Update permission checker function to run only after the user role has been retrieved from the database

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_front.git`
- Switch to the project directory - `cd travel_tool_front`
- Checkout - `git checkout bg-dashboard-redirect-161755766`
- Start the app - `make start`
- Start the backend of this project - `git clone https://github.com/andela/travel_tool_back.git && make start`
- Navigate to travela in your browser and log in
- Navigate to the dashboard page
- Refresh the page or hit `cmd + R` to refresh the page
- The dashboard page should be refreshed properly without redirecting to the request page

#### Any background context you want to provide?
- Ensure that you have the role of `Travel Administrator`;

#### What are the relevant pivotal tracker stories?
[#161755766](https://www.pivotaltracker.com/story/show/161755766)

#### Screenshots (if appropriate)
N/A

#### Questions:
